lib/rules/no-duplicate-landmark-labels.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Duplicate label on <${element}> detected. If two of the same landmark elements or landmark roles are found in the same template, ensure that they have a unique label (provided by the aria-label or aria-labelledby attribute).`;
+}
+
+const ROLE_LANDMARK_MAP = {
+  banner: 'header',
+  main: 'main',
+  complementary: 'aside',
+  form: 'form',
+  search: 'form',
+  navigation: 'nav',
+  contentinfo: 'footer',
+};
+
+const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
+
+const LANDMARK_ROLES = Object.keys(ROLE_LANDMARK_MAP);
+
+module.exports = class NoDuplicateLandmarkLabels extends Rule {
+  constructor(options) {
+    super(options);
+    this._landmarkHash = [];
+  }
+
+  visitor() {
+    return {
+      ElementNode(node) {
+        const hasRoleAttribute = AstNodeInfo.hasAttribute(node, 'role');
+
+        // Return if it is not a landmark element or doesn't have role
+        if (!(hasRoleAttribute || LANDMARK_ELEMENTS.includes(node.tag))) {
+          return;
+        }
+
+        const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
+
+        // <nav>
+        const isLandMarkElement = LANDMARK_ELEMENTS.includes(node.tag);
+
+        // <div role="navigation">
+        const isLandMarkRole = LANDMARK_ROLES.includes(roleValue);
+
+        if (isLandMarkElement || isLandMarkRole) {
+          // check for label
+          const labelValue = AstNodeInfo.elementAttributeValue(node, 'aria-label') || '';
```

Comment:
The code below doesn't seem to account for `labelValue ` to be anything other than a string, but in reality it can be either a string or AST node.

Reply:
is this a good example of what you are talking about?
```hbs
<div role="navigation" aria-label={{myLabel}}></div>
<nav aria-label={{myLabel}}></div>
```

Reply:
@rwjblue If we are thinking about going that way, I think we have a lot of ground to cover. 
There are so many variations like:
```hbs
<main aria-label={{myLabel}}></main>
<main aria-label="label-{{if myLabel "primary" "secondary"}}></main>
<main aria-label={{@myLabel}}></main>
<main aria-label={{concat "label-" myLabel}}></main>
```
I think the best possible way here, is to convert all the AST Node values to their stringified forms 
so that it will be easy to compare against. Is there any util available already in the codebase for the same?
I took a look at this one 
https://github.com/ember-template-lint/ember-template-lint/blob/cb218e3404bc865434c8dae537aaa4982464fb58/lib/helpers/ast-node-info.js#L110
But it seems not enough to handle other than `TextNode`, for `MustacheStatement` I can get a value like `{{myLabel}}` but for others like `ConcatStatement` we need to handle I guess..

Reply:
In theory, it should be possible to add support for other types of values but we don't have any good examples of that in the codebase at the moment. Basically, there is no particular reason that we can't tell these are the same value:

```
<main aria-label={{myLabel}}></main>
<main aria-label={{myLabel}}></main>
```

Like you said, we'd need to represent the value independently for each. Perhaps we could use the `sourceForNode` to compare the raw source :thinking:. We'd have to account for this (but taht isn't _too_ hard):

```hbs
<main aria-label={{myLabel}}></main>
{{#let (concat "special-" this.otherThing) as |myLabel|}}
  <main aria-label={{myLabel}}></main>
{{/let}}
```

---

I think for a first pass, we can just bail on checking if the the value is _not_ a TextNode. In my experience they are _very commonly_ literal values, so it makes sense to start with checking them that way, and return early (without logging) when we see a dynamic value. Though I'm super happy to help work out the more complicated thing if you would like to do that instead.

Reply:
Ok, Sound good to me,  hence, for now, I will return straight-away if the role value type is not `TextNode`. And I will address the complicated stuff for dynamic values in a separate PR